src/main/java/frc/bionic/swerve/MotorFaclon500.java

```diff
@@ -0,0 +1,183 @@
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+/**
+ * MotorFaclon500
+ */
+public class MotorFaclon500 implements IMotor{
+    
+    //TODO test values
+    // Initial, default PID constants, overridden by persistent shuffleboard fields
+    private static final double kMotorP = 0.0;
+    private static final double kMotorI = 0.0;
+    private static final double kMotorD = 0.0;
+    private static final double kMotorIz = 0.0;
+    private static final double kMotorFf = 0.0;
+    private static final int kMotorSlot = 0;
+
+    /**
+    To keep the motor closer to peek power, we limit the max output.
+    See torque/speed curves on https://motors.vex.com/  
+    */ 
+    private static final double kMotorMax = 0.7;
+    private static final double kMotorMin = -0.7;
+
+    //Devices, Sensors, Actuators
+    private MedianFilter velAverage; //For displaying average RPM
+    private TalonFX motor;
+
+    //Shufflebord Related
+    private String            name;
+    private String            shuffleboardTabName;
+  
+    NetworkTableEntry         sb_vel_rpm;
+    NetworkTableEntry         sb_vel_avg;
+    NetworkTableEntry         sb_vel_rpm_set;
+    NetworkTableEntry         sb_vel_apply;
+  
+    NetworkTableEntry         sb_pid_kp;
+    NetworkTableEntry         sb_pid_ki;
+    NetworkTableEntry         sb_pid_kd;
+    NetworkTableEntry         sb_pid_kiz;
+    NetworkTableEntry         sb_pid_kff;
+    NetworkTableEntry         sb_pid_max;
+    NetworkTableEntry         sb_pid_min;
+    NetworkTableEntry         sb_pid_apply;
+
+    public MotorFaclon500(int deviceId, String name, String shuffleboardTabName){
+
+        //Save parameter values to be used elsewhere
+        this.name = name;
+        this.shuffleboardTabName = shuffleboardTabName;
+
+        //Makes a new Talon motor
+        motor = new TalonFX(deviceId);
+        
+        //Resets the motor to default
+        motor.configFactoryDefault();
+
+        //Initially set RPM to 0 (Stopped Motor)
+        setGoalRPM(0.0);
+
+        //Prepare to display velocity on shuffleboard
+        velAverage = new MedianFilter(50);
+
+        motor.config_kP(kMotorSlot, kMotorP);
+        motor.config_kI(kMotorSlot, kMotorI);
+        motor.config_kD(kMotorSlot, kMotorD);
+        motor.config_IntegralZone(kMotorSlot, kMotorIz);
+        motor.config_kF(kMotorSlot, kMotorFf);
+        motor.configPeakOutputForward(kMotorMax);
+        motor.configPeakOutputReverse(kMotorMin);
+
+        //Initilize Shuffleboard Interface
+        initShuffleboard();
+        
+    }
+
+    //Interface Implementation
+    public void setGoalRPM(double goalRPM){
+        SmartDashboard.putNumber(name + " goal", goalRPM);
+        motor.set(TalonFXControlMode.Velocity, goalRPM);
+    }
+
+    //Interface Implementation
+    public double getVelocityRPM(){
+        return motor.getActiveTrajectoryVelocity(); //THIS IS PER 100MS, change if needed in RPM
```

Comment:
Yes, the implementation requires RPM so that all implementations of the interface provide consistent values. (The higher level code will be expecting RPM.)